index.xml

```diff
@@ -603,6 +645,7 @@ be generated as:
     <figure>
      <artwork>
 Signature: keyId="rsa-key-1",algorithm="rsa-sha256",
```

Comment:
I suggest that this is a breaking change and so we should create a new "algorithm" for this... something like "rsasig-2019", and then define that to use something like "rsa + sha256". We should probably also define an Ed25519 equivalent "ed25519sig-2019", which uses eddsa + sha256. This enables us to keep backwards compatibility with rsa-sha256 (because the old implementations will silently fail w/ the new optional parameters.

Reply:
The question is whether we want to version algorithms or signatures, namely the string to sign. 

I'd stick with the current algos naming, so that people can easily reference them. And add another, mandatory signed field named `v` usable to specify the version.

This allows our users to deprecate older versions of the spec using a standard, versioned parser. Eg.

```
Signature: v=1.1; ... 

# I require (2> v > 1.5) 
```

Instead of having them to search in their algorithm's repo.

Reply:
IIRC, we want to discourage the use of the `algorithm` parameter in favor of getting this information from the key instead: https://github.com/w3c-dvcg/http-signatures/blob/43577c3e6debd30ba3fc56c3bfe4c783f3a059f9/index.xml#L218-L228

Also, the `algorithm` parameter is primarily about the crypto rather than building the signature string, as @ioggstream says. So it sounds like we want some other field here (version or similar).